File: leetCode/fukun/101-200/lc_167.js
const twoSum = function (numbers, target) {
    let curr1 = 0;
    let curr2 = numbers.length - 1;
    while (curr1 < curr2) {
        if (numbers[curr1] + numbers[curr2] > target) {
            curr2--;
            continue;
        }
        if (numbers[curr1] + numbers[curr2] < target) {
            curr1++;
            continue;
        }
        return [curr1 + 1, curr2 + 1];
    }
};
console.log(twoSum([2, 7, 11, 15], 9));

File: leetCode/lyc/simple/lc_167.java
/*two sumII-input array is sorted*/
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] res=new int[2];
        if(numbers.length<2){
            return res;
        }
        int i=0;
        int j=numbers.length-1;
        while(i<j){
            if(numbers[i]+numbers[j]==target){
                res[0]=i+1;
                res[1]=j+1;
                break;
            }
            else if(numbers[i]+numbers[j]>target){
                j--;
            }else{
                i++;
            }
        }
        return res;
    }
}